libsolidity/Parser.cpp

```diff
@@ -622,6 +622,16 @@ ASTPointer<Statement> Parser::parseStatement()
 		statement = nodeFactory.createNode<Return>(expression);
 		break;
 	}
+	case Token::Throw:
+	{
+		ASTNodeFactory nodeFactory(*this);
+		ASTPointer<Expression> expression;
+		if (m_scanner->next() != Token::Semicolon)
+			BOOST_THROW_EXCEPTION(createParserError("Throw statement cannot have parameters."));
```

Comment:
Perhaps do not assume that the next token is a parameter. This error message given for `throw }` would sound weird.